service/history/hsm/registry.go

```diff
@@ -140,6 +143,27 @@ func RegisterImmediateExecutor[T Task](r *Registry, executor ImmediateExecutor[T
 	return nil
 }
 
+// RegisterRemoteMethod registers an [RemoteExecutor] for the given remote method definition.
+// Returns an [ErrDuplicateRegistration] if an executor for the type has already been registered.
+func RegisterRemoteMethod[I any, O any, R RemoteMethod[I, O]](r *Registry, executor RemoteExecutor[I, O]) error {
+	var method R
+	methodName := method.Name()
```

Comment:
This limits the use pointer receivers for implementations of `R` as you may get NPEs.
I see this mistake was copied from the task registration functions, and I missed this in a previous review.
I'll fix those later.

As I mentioned in the previous path, you should make `MethodDefinition` non generic or you'll have to use reflection excessively when using them.